c7n/tags.py

```diff
@@ -432,11 +432,15 @@ def process(self, resources):
 
     def process_resource_set(self, client, resource_set, tags):
         mid = self.manager.get_model().id
-        self.manager.retry(
-            client.create_tags,
-            Resources=[v[mid] for v in resource_set],
-            Tags=tags,
-            DryRun=self.manager.config.dryrun)
+        try:
+            self.manager.retry(
+                client.create_tags,
+                Resources=[v[mid] for v in resource_set],
+                Tags=tags,
+                DryRun=self.manager.config.dryrun)
+        except ClientError as e:
+            if e.error['Code'] == 'InvalidSnapshot.NotFound':
+                pass
```

Comment:
this would block the creation of tags for any resource if one isn't found, this should remove the missing snapshot, regardless of that though, we shouldn't be calling out specific errors here for a given resource type, rather delegate that to the query resource manager's retry function